modules/dnn/src/layers/resize_layer.cpp

```diff
@@ -110,7 +135,9 @@ class ResizeLayerImpl : public ResizeLayer
             scaleWidth = static_cast<float>(inputs[0].size[3]) / outWidth;
     }
 
-    void forward(InputArrayOfArrays inputs_arr, OutputArrayOfArrays outputs_arr, OutputArrayOfArrays internals_arr) CV_OVERRIDE
+
+    template <typename Tmat>
```

Comment:
Create a separate method `forward_ocl` which covers OpenCL implementation:
```
if ((interpolation == "nearest" && !alignCorners && !halfPixelCenters) || (interpolation == "opencv_linear" && depth != CV_8S) ||
            (interpolation == "bilinear" && halfPixelCenters && depth != CV_8S))
```

Reply:
Alright. Thx!

Reply:
@kallaballa this is still actual. I think method `forward_ocl` is better than introducing a template func.

Reply:
@dkurt my bad. already had it... fixing tomorrow